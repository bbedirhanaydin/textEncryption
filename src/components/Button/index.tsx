import React from 'react';
import {StyleSheet, Pressable, Text, Dimensions} from 'react-native';

const Button = ({onPress, content}: any) => {
  return (
    <Pressable testID="onpress-trigger" onPress={onPress} style={styles.button}>
      <Text testID="button-content" style={styles.text}>
        {content}
      </Text>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  button: {
    width: Dimensions.get('window').width / 2,
    backgroundColor: '#009FBD',
    borderRadius: 5,
    padding: 5,
  },
  text: {
    fontSize: 25,
    color: '#FFFFFF',
    textAlign: 'center',
  },
});

export default Button;
