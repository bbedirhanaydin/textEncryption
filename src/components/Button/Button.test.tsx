import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import Button from '.';

test('should match with snapshot', () => {
  const comp = render(<Button />);
  expect(comp).toMatchSnapshot();
});

test('should render content correctly', () => {
  const testContent = 'test';
  const comp = render(<Button content={testContent} />);
  const buttonText = comp.getByTestId('button-content').children[0];
  expect(buttonText).toBe(testContent);
});

test('should triger onPress', () => {
  const mockFunction = jest.fn();
  const comp = render(<Button onPress={mockFunction} />);

  const buttonTouchable = comp.getByTestId('onpress-trigger');
  fireEvent(buttonTouchable, 'press');

  expect(mockFunction).toBeCalled;
});
