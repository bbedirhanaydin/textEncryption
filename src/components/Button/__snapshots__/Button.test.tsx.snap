// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#009FBD",
      "borderRadius": 5,
      "padding": 5,
      "width": 375,
    }
  }
  testID="onpress-trigger"
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontSize": 25,
        "textAlign": "center",
      }
    }
    testID="button-content"
  />
</View>
`;
